library/ssl_cli.c

```diff
@@ -3090,8 +3090,10 @@ static int ssl_parse_server_key_exchange( mbedtls_ssl_context *ssl )
 #endif
         mbedtls_md_type_t md_alg = MBEDTLS_MD_NONE;
         mbedtls_pk_type_t pk_alg = MBEDTLS_PK_NONE;
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2)
```

Comment:
TLS 1.2 only function, what about to rather guard the whole function with MBEDTLS_SSL_PROTO_TLS1_2 and to remove the MBEDTLS_SSL_PROTO_TLS1_2 guards inside the function. Otherwise probably better to do the same with other TLS 1.2 specific functions in this file. It would be maybe even faster and simpler to guard with MBEDTLS_SSL_PROTO_TLS1_2 all the functions of the file but the one we are using in TLS 1.3.

Reply:
Guard whole file should be better choice.